packages/payment-processor/src/payment/encoder-payment.ts

```diff
@@ -41,10 +43,21 @@ export function encodeRequestPaymentWithoutSwap(
         ...overrides,
       };
     case ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_FEE_PROXY_CONTRACT:
-      return {
-        ...prepareErc20FeeProxyPaymentTransaction(request, amount, feeAmount),
-        ...overrides,
-      };
+      if (options?.isEscrow) {
```

Comment:
I'm not comfortable with that choice. I think escrow is, for now, a minor option, and later probably a separate RN, so I think it's not worth it re-using the same `encodeRequestPayment` method, but instead make a specific one like `encodeRequestPaymentWithEscrow`.